flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,235 @@ import Sentry
 import UIKit
 
 public class SwiftSentryFlutterPlugin: NSObject, FlutterPlugin {
-  public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "sentry_flutter", binaryMessenger: registrar.messenger())
-    let instance = SwiftSentryFlutterPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
-  }
-
-  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    result("iOS " + UIDevice.current.systemVersion)
-  }
-
-  override init() {
-    if let path = Bundle.main.path(forResource: "Info", ofType: "plist") {
-      if let resource = NSDictionary(contentsOfFile: path) {
-        if let dsn = resource.object(forKey: "SentryDsn") {
-          SentrySDK.start { options in
-              options.dsn = dsn as? String
-              options.debug = resource["SentryDebug"] as? Bool ?? false
-              options.attachStacktrace = true
-          }
-        }
-      }
-    }
-  }
+
+    var sentryOptions: Options?
+
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let channel = FlutterMethodChannel(name: "sentry_flutter", binaryMessenger: registrar.messenger())
+        let instance = SwiftSentryFlutterPlugin()
+        registrar.addMethodCallDelegate(instance, channel: channel)
+    }
+
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        switch call.method as String{
+            case "deviceInfos" :
+              deviceInfos(result: result)
+              break
+
+            case "initNativeSdk" :
+              initNativeSdk(call, result: result)
+              break
+
+        case "captureEnvelope" :
+            captureEnvelope(call, result: result)
+            break
+        default:
+            result(FlutterMethodNotImplemented)
+        }
+
+
+    }
+
+    private func deviceInfos(result: @escaping FlutterResult){
+        SentrySDK.configureScope{ scope in
+            let serializedScope = scope.serialize()
+            let context = serializedScope["context"]
+
+            var infos = ["contexts":context]
+            // TODO DEBUG context
+
+            if let integrations = self.sentryOptions?.integrations {
+                infos["integrations"] = integrations
+            }
+
+            // TODO get the sdkVersion from SDK
+            infos["package"] = ["version": "6.0.9" ,"sdk_name":"cocoapods:sentry-cocoa"]
```

Comment:
@philipphofmann is there a way to get the SDK version? so we don't need to hardcode

Reply:
@rxlabz as far as I can tell `SentryMeta.versionString` would work?

Reply:
`SentryMeta` doesn't seems public

Reply:
Yes, exactly @marandaneto. Craft calls the [VersionBump](https://github.com/getsentry/sentry-cocoa/blob/master/Utils/VersionBump/main.swift) script to update `SentryMeta` when we do a new release.

Reply:
@philipphofmann yep, we'll get there, but the question here is different, how do we read the Cocoa version on runtime?

Reply:
There is no straightforward way to do this. We could easily add a way to retrieve this, but why do we even need this? I don't see something like this in react-native: https://github.com/getsentry/sentry-react-native/blob/master/ios/RNSentry.m

Reply:
@philipphofmann to populate the `sdk.packages`, not all the other SDKs do it, but they should.
so when there's an event, we know which SDKs were involved, eg;

an event captured by Dart SDK, only has the Dart package.
an event captured by the Flutter SDK, has the Dart SDK, Flutter SDK and Native SDK (Android or iOS) as they are the ones sending over the network.

Reply:
Okay then making this public isn't a problem. Should we just make `SentryMeta` public or should we add methods to `SentrySDK`?

Reply:
@philipphofmann I like how Android Does, we have a field on options, `SentryOptions.sdkVersion`, on SDK init, we set the values, so it's possible to read thru the `OptionsConfiguration` callback.

```
SentryAndroid.init(options-> {
  // options.sdkVerrsion.name and version
})
```

Reply:
I just opened up a PR for this https://github.com/getsentry/sentry-cocoa/pull/859